src/mmrelay/main.py

```
@@ -322,44 +356,378 @@ def signal_handler() -> None:
+        The callable executes on a daemon thread so shutdown can continue even if
+        the step hangs. Exceptions are logged and swallowed so remaining cleanup
+        still runs.
+        """
+        loop = asyncio.get_running_loop()
+        step_result: asyncio.Future[Exception | None] = loop.create_future()
+
+        def _run_step() -> None:
+            step_error: Exception | None = None
+            try:
+                step_func()
+            except Exception as exc:
```

Ruff (0.15.6)

[warning] Do not catch blind exception: `Exception`

(BLE001)